schema/schema_merge.go

```diff
@@ -205,6 +210,31 @@ func (m *SchemaMerger) SchemaForModule(meta *module.Meta) (*schema.BodySchema, e
 			if err == nil {
 				mergedSchema.Blocks["module"].DependentBody[schema.NewSchemaKey(depKeys)] = depSchema
 			}
+
+			// There's likely more edge cases with how source address can be represented in config
+			// vs in module manifest, but for now we at least account for the common case of TF Registry
+			if strings.HasPrefix(module.SourceAddr, "registry.terraform.io/") {
```

Comment:
Do we want to have two `depSchema` entries (one with prefix from above and this without) for modules from the TF registry? Or could we strip the prefix like this without duplicating the block?

```go
sourceAddr := module.SourceAddr
// There's likely more edge cases with how source address can be represented in config
// vs in module manifest, but for now we at least account for the common case of TF Registry
if strings.HasPrefix(sourceAddr, "registry.terraform.io/") {
    sourceAddr = strings.TrimPrefix(sourceAddr, "registry.terraform.io/")
}
```

Reply:
so AFAIK the full address is also valid, even though the vast majority will use the short one, so I would keep both

Reply:
Okay, that makes sense. Thanks for the clarification.